src/main/java/net/fabricmc/loader/launch/knot/MixinStringPropertyKey.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2016 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.loader.launch.knot;
+
+import org.spongepowered.asm.service.IPropertyKey;
+
+import java.util.Objects;
+
+public class MixinStringPropertyKey implements IPropertyKey {
+	public final String key;
+
+	public MixinStringPropertyKey(String key) {
+		this.key = key;
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (!(obj instanceof MixinStringPropertyKey)) {
```

Comment:
Shortcircuit referential equality
```suggestion
		if (this == obj) {
				return true;
		}
		if (!(obj instanceof MixinStringPropertyKey)) {
```

Reply:
Not necessary, but a good improvement

Reply:
Unfortunately, change suggestions are not that extensive, so that change will have to be done directly.